### PR TITLE
Pull request #13: enable 'frege-native-gen -t generte.all'

### PR DESCRIPTION
**Pull request by jwausle**
Enhance ‚frege-native-gen‘ cmdline:

$ frege-native-gen -c class1 -c class2  # to generate >1 class
$ frege-native-gen -t generate.all         # to generate all classen from generate.all as types.properties
$ frege-native-gen                                # same as above with default: types.properties

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -37,15 +37,15 @@ run {
 
 defaultTasks "build"
 
-sourceCompatibility = 1.7
-targetCompatibility = 1.7
+sourceCompatibility = 1.8
```
Comment (mmhelloworld) [inline build.gradle]: Why is this needed? We should be able to support 1.7 as other Frege projects are still on 1.7.

File: src/main/frege/frege/nativegen/Main.fr [hunk 2]
```diff
@@ -170,14 +175,14 @@ parseArgs args = go optDefault args where
...
-    = go optSoFar.{name = className, category = Class} rest
+    = go optSoFar.{name = optSoFar.name ++ (if optSoFar.name == "" then "" else ",") ++ className, category = Class} rest
   go optSoFar ("-d" : directory : rest)
     = go optSoFar.{outputDir = directory} rest
   go optSoFar ("-t" : typesFilePath : rest)
     = go optSoFar.{typesFilePath = typesFilePath} rest
   go _ (invalidOption:_) = Left $ "Invalid option: " ++ invalidOption
 
-data Category = Package | Class
+data Category = Package | Class | Types
```
Comment (mmhelloworld) [inline src/main/frege/frege/nativegen/Main.fr]: I think we don't need a new category. We should instead improve `Class` to take multiple values.

File: build.gradle (hunk 1 above)
Comment (jwausle) [inline build.gradle]: It dosen`t matter. ... Anyone while the development I had compilation or runtime error. I checked this again with source- and target-level=1.7. It works.

File: src/main/frege/frege/nativegen/Main.fr (hunk 2 above)
Comment (jwausle) [inline src/main/frege/frege/nativegen/Main.fr]: I want to keep the change minimal inverse. Maybe I interprete the catagory different than Yours. For me it was the simplest way to catch a new case in the main function.

Comment (jwausle) [inline src/main/frege/frege/nativegen/Main.fr]: Hint: I improved the Class type too. You can use it like that:

$ frege-native-gen -c one.class -c another.class -c so.many.classes -c you.want ...

File: build.gradle (hunk 1 above)
Comment (mmhelloworld) [inline build.gradle]: can we change it to 1.7 please then?

File: src/main/frege/frege/nativegen/Main.fr (hunk 2 above)
Comment (mmhelloworld) [inline src/main/frege/frege/nativegen/Main.fr]: I would rather remove the category `Types` and just use `Class` and keep the option as `-c class1,class2,class3`

**Comment by mmhelloworld** [comment]
Thanks! I have merged your pull request after making a few changes that I mentioned in the comments.

File: build.gradle (hunk 1 above)
Comment (jwausle) [inline build.gradle]: Yes You can change to 1.7.

**Comment by jwausle** [comment]
Your welcome and thanks too:)
